vendor/github.com/go-logr/logr/logr.go

```diff
@@ -14,18 +14,15 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Package logr defines abstract interfaces for logging.  Packages can depend on
-// these interfaces and callers can implement logging in whatever way is
-// appropriate.
-//
 // This design derives from Dave Cheney's blog:
 //     http://dave.cheney.net/2015/11/05/lets-talk-about-logging
```

Comment:
https://dave.cheney.net/2017/01/26/context-is-for-cancelation 🙃

Reply:
I know that this controversial, which is also why this hasn't been decided for Kubernetes yet.

But in the meantime the world is moving on and context is getting used to propagate loggers, for example in controller-runtime (initially with a custom solution, now with that new logr API -> https://github.com/kubernetes-sigs/controller-runtime/issues/1275). It's simply easier to attach another key to a context than it is to add a new parameter to each function.

Reply:
That is disappointing. Besides code smell arguments, contexts that use more WithValue are expensive to use regularly. It's a linked list with linear search by way of recursive interface calls ...

Reply:
agreed... I wouldn't be pleased with use of this within k8s.io/kubernetes to plumb a logger, or to frequently get one